lib/utils/logger.js

```diff
@@ -147,37 +136,31 @@ function setLogFilePath(path) {
 }
 
 /**
- * Log a message to console.
+ * Log a message to console in accordance with the SDK configuration
  *
- * If an Error object is passed as the message, it will log both the error's
+ * NOTE: If an Error object is passed as the message, it will log both the error's
  * message and stack trace.
  *
  * @param {string} message Log message
- * @param {string} level Log level
+ * @param {object} options Options object
+ * @param {string} options.level Log level
+ * @param {object} options.meta Optional log metadata
  *
- * @returns {winston.log|null} Returns a log message if the {@link module:config} option `verbose` is `true`;
- * otherwise, returns `null`.
+ * @returns {winston.log|null} Returns a log message, along with optional metadata
+ * (if the {@link module:config} option `verbose` is `true`).
  */
-function log(message, level) {
+function log(message, { level = 'info', meta = {} } = {}) {
```

Comment:
Question: does the quiet flag affect the `log` function?

Reply:
Good question. This suggests that I should add a test for the effects of the 2 config props. Stand by…

Reply:
(The answer is it doesn't — directly. Rather, it simply changes the winston log level, which dictates whether or not a log is printed.)

Reply:
Tests added